lib/chai/utils/hasProperty.js

```diff
@@ -35,30 +33,19 @@ var type = require('type-detect');
  *     hasProperty(2, obj.arr);  // true
  *     hasProperty(3, obj.arr);  // false
  *
+ * @param {String|Symbol} name
  * @param {Object} object
- * @param {String|Number} name
  * @returns {Boolean} whether it exists
  * @namespace Utils
- * @name getPathInfo
+ * @name hasProperty
  * @api public
  */
 
-var literals = {
-    'number': Number
-  , 'string': String
-};
-
 module.exports = function hasProperty(name, obj) {
-  var ot = type(obj);
-
-  // Bad Object, obviously no props at all
-  if(ot === 'null' || ot === 'undefined')
+  if (obj === undefined || obj === null) {
```

Comment:
We lose a little bit of safety net here from using `obj === undefined` instead of `typeof obj === 'undefined'`, in case the user has done terrible things and redefined `undefined`. However, I'm not sure if this safety net is actually necessary, and I don't have time at the moment to check what we do elsewhere in the code.